include/internal/list.h

```diff
@@ -61,10 +61,10 @@
     static ossl_unused ossl_inline void                                     \
     ossl_list_##name##_remove(OSSL_LIST(name) *list, type *elem)            \
     {                                                                       \
-        if (list->head == elem)                                             \
```

Comment:
(Above) I think it's quite undesirable as a design choice that the name of the list node field in a structure be bound to have a fixed relationship to the name of the list accessor functions...

I think it would be better to add another argument to the macro for the field name and drop the `ossl_list_` prefix on the field names altogether.

Since these are statics, we can also drop the `ossl_` prefix.

The usage I have more in mind is (for the most common cases):
```c
struct quic_urxe_st {
  OSSL_LIST_MEMBER(node, QUIC_URXE);
};
DEFINE_LIST_OF(QUIC_URXE, QUIC_URXE, node);
OSSL_LIST(QUIC_URXE) urxe_list;
list_QUIC_URXE_remove(&urxe_list, urxe);
```

Reply:
The list node field name is entirely internal, it's name is not relevant outside of list.h.  The only requirement is that it must never clash with other structure fields or other list names.

The former is possible with any unlikely prefix (which `ossl_list_` is in this context, as you note).
The latter is why I've included the list name as part of it.

Your usage example matches the pattern I'm using, apart from the `node` name.  The only reason to include the node name is to allow direct access to the list's internal fields which I think is a very bad thing (which you disagree with).